theano/tensor/basic.py

```diff
@@ -5888,41 +5888,79 @@ def all(x, axis=None, keepdims=False):
 class Diagonal(Op):
     """Return specified diagonals.
 
+    Usage: Diagonal(offset, axis1, axis2)(x)
+
+    `x` has to be a tensor with 2 or more dimensions.
+
     Parameters
     ----------
-    x
+    offset : int
+        Indicates which diagonal to extract, `0` as main diagonal, positive as
+        diagonals in the upper triangle, and vice versa.
+
+    axis1, axis2 : int
+        Indicates on which dimensions to extract the diagonals.
+
+    x: symbolic tensor
         A tensor variable with x.ndim >= 2.
 
     Returns
     -------
-    vector
+    vector : symbolic vector
         A vector representing the diagonal elements.
 
     """
     __props__ = ("offset", "axis1", "axis2")
 
     def __init__(self, offset=0, axis1=0, axis2=1):
+        self.view = False
         if numpy_diagonal_return_view:
             self.view_map = {0: [0]}
+            self.view = True
         self.offset = offset
         self.axis1 = axis1
         self.axis2 = axis2
 
     def make_node(self, x):
         x = as_tensor_variable(x)
-        assert x.ndim >= 2
+        if x.ndim < 2:
+            raise TypeError('Diagonal needs an input with 2 or more '
+                            'dimensions', x)
         return Apply(self, [x], [tensor(dtype=x.dtype,
                                         broadcastable=[False] * (x.ndim - 1))])
 
     def perform(self, node, inputs, outputs):
         (x,) = inputs
         (z,) = outputs
-        z[0] = x.diagonal(self.offset, self.axis1, self.axis2)
+        if self.offset != 0 or self.axis1 != 0 or self.axis2 != 1:
+            raise ValueError('Currently Diagonal doesn\'t support non-default'
```

Comment:
We talked about it in private. For the record, this breaks the CPU interface because the op used to work for non default values. Now it raises a ValueError.